pkg/kubelet/network/kubenet/kubenet_linux_test.go

```diff
@@ -232,37 +232,55 @@ func TestGenerateMacAddress(t *testing.T) {
 // TestInvocationWithoutRuntime invokes the plugin without a runtime.
 // This is how kubenet is invoked from the cri.
 func TestTearDownWithoutRuntime(t *testing.T) {
-	fhost := nettest.NewFakeHost(nil)
-	fhost.Legacy = false
-	fhost.Runtime = nil
-	mockcni := &mock_cni.MockCNI{}
-
-	fexec := &exec.FakeExec{
-		CommandScript: []exec.FakeCommandAction{},
-		LookPathFunc: func(file string) (string, error) {
-			return fmt.Sprintf("/fake-bin/%s", file), nil
+	testCases := []struct {
+		podCIDR  string
+		ip       string
+		expectIP string
+	}{
+		{
+			podCIDR:  "10.0.0.1/24",
+			ip:       "10.0.0.1",
+			expectIP: "10.0.0.1/32",
+		},
+		{
+			podCIDR:  "2001:beef::1/48",
+			ip:       "2001:beef::1",
+			expectIP: "2001:beef::1/128",
 		},
 	}
+	for _, tc := range testCases {
+		fhost := nettest.NewFakeHost(nil)
+		fhost.Legacy = false
+		fhost.Runtime = nil
+		mockcni := &mock_cni.MockCNI{}
+
+		fexec := &exec.FakeExec{
+			CommandScript: []exec.FakeCommandAction{},
+			LookPathFunc: func(file string) (string, error) {
+				return fmt.Sprintf("/fake-bin/%s", file), nil
+			},
+		}
 
-	kubenet := newFakeKubenetPlugin(map[kubecontainer.ContainerID]string{}, fexec, fhost)
-	kubenet.cniConfig = mockcni
-	kubenet.iptables = ipttest.NewFake()
+		kubenet := newFakeKubenetPlugin(map[kubecontainer.ContainerID]string{}, fexec, fhost)
+		kubenet.cniConfig = mockcni
+		kubenet.iptables = ipttest.NewFake()
 
-	details := make(map[string]interface{})
-	details[network.NET_PLUGIN_EVENT_POD_CIDR_CHANGE_DETAIL_CIDR] = "10.0.0.1/24"
-	kubenet.Event(network.NET_PLUGIN_EVENT_POD_CIDR_CHANGE, details)
+		details := make(map[string]interface{})
+		details[network.NET_PLUGIN_EVENT_POD_CIDR_CHANGE_DETAIL_CIDR] = tc.podCIDR
+		kubenet.Event(network.NET_PLUGIN_EVENT_POD_CIDR_CHANGE, details)
 
-	existingContainerID := kubecontainer.BuildContainerID("docker", "123")
-	kubenet.podIPs[existingContainerID] = "10.0.0.1"
+		existingContainerID := kubecontainer.BuildContainerID("docker", "123")
+		kubenet.podIPs[existingContainerID] = tc.ip
```

Comment:
This adds an IP to the podIPs, then in the teardown function, the podIP is checked for and deleted. This test function checks that the DelNetwork is called correctly.